packages/relay/src/lib/clients/cache/redisCache.ts

```diff
@@ -160,6 +160,38 @@ export class RedisCache implements ICacheClient {
     this.logger.trace(`${requestIdPrefix} caching multiple keys via ${callingMethod}, total keys: ${entriesLength}`);
   }
 
+  /**
+   * Stores multiple key-value pairs in the cache using pipelining.
+   *
+   * @param keyValuePairs - An object where each property is a key and its value is the value to be cached.
+   * @param callingMethod - The name of the calling method.
+   * @param {number} [ttl] - The time-to-live (expiration) of the cache item in milliseconds.
+   * @param requestIdPrefix - Optional request ID prefix for logging.
+   * @returns {Promise<void>} A Promise that resolves when the values are cached.
+   */
+  async pipelineSet(
+    keyValuePairs: Record<string, any>,
+    callingMethod: string,
+    ttl?: number | undefined,
+    requestIdPrefix?: string,
+  ): Promise<void> {
+    const resolvedTtl = (ttl ?? this.options.ttl) / 1000; // convert to seconds
+
+    const pipeline = this.client.multi();
```

Comment:
Q: are there any memory impacts of this get?
Does it initiate a new one each time or should we consider getting one and setting as a class variable that is used in method?

Also do we need to clear the pipeline after or does the API do it for you?

Reply:
I believe closing the pipeline is handled by redis implementation itself, we don't need to do anything. New one is opened upon new request. 
Also I don't think that we can use it as class variable, because when execution is completed, the pipeline is closed and until you execute the request it will just keep storing your commands (set,get,etc) and not sending them over. 
Here is comment from the go-redis implementation, but should be the same for node-redis. https://github.com/redis/go-redis/discussions/2583#discussioncomment-5823938